File: Aplikacja/Serwer/CsharpServerApp/oto-auto-c-sharp-server/oto-auto-c-sharp-server/Logic/Offers/Models/OfferCardComponentModel.cs
namespace oto_auto_c_sharp_server.Logic.Offers.Models;

public class OfferCardComponentModel
{
    public int OfferId { get; set; }
    public OfferImage OfferMainImage { get; set; }
    public string OfferTitle { get; set; }
    public string YearOfProduction { get; set; }
    public string Mileage { get; set; }
    public string MileageUnit { get; set; }
    public string FuelType { get; set; }
    public string EngineCapacity { get; set; }
    public string OfferPrice { get; set; }
    public string OfferCurrency { get; set; }
}

File: Aplikacja/ReactApp/oto-auto/src/redux/model/offer-card-component.model.ts
import { VehicleDetailsRow } from "../../models/vehicle-details-row";

export interface OfferCardComponentModel {
    offerId: number;
    offerMainImage: OfferImage,
    offerTitle: string,
    yearOfProduction: string,
    mileage: string,
    mileageUnit: string,
    fuelType: string,
    engineCapacity: string,
    offerPrice: string,
    offerCurrency: string
}

export interface OfferActivityComponentModel {
    offerId: number;
    offerImages: OfferImage[],
    offerTitle: string,
    offerDescription: string
    yearOfProduction: string,
    mileage: string,
    fuelType: string,
    engineCapacity: string,
    offerPrice: string,
    offerCurrency: string
    equipments: string[]
    vehicleAttributes: VehicleDetailsRow[]
}

export interface OfferCardComponentStateModel {
    offerCardsComponent: OfferCardComponentModel[],
    offer: OfferActivityComponentModel | null
    offerImages: OfferImage[]
    redisNotResponding: boolean
}

export interface OfferImage {
    imageBytes: string,
    isMainImage: boolean
}